src/plugins/orangehrmCoreOAuthPlugin/test/Api/OAuthClientAPITest.php

```diff
@@ -57,7 +58,18 @@ public function testUpdate(TestCaseParams $testCaseParams): void
         $this->createKernelWithMockServices([Services::AUTH_USER => $this->getMockAuthUser($testCaseParams)]);
 
         $this->registerServices($testCaseParams);
-        $api = $this->getApiEndpointMock(OAuthClientAPI::class, $testCaseParams);
+        $api = $this->getApiEndpointMockBuilder(
+            OAuthClientAPI::class,
+            [
+                RequestParams::PARAM_TYPE_ATTRIBUTE => $testCaseParams->getAttributes() ?? [],
+                RequestParams::PARAM_TYPE_QUERY => $testCaseParams->getQuery() ?? [],
+                RequestParams::PARAM_TYPE_BODY => $testCaseParams->getBody() ?? [],
+            ]
+        )->onlyMethods(['generateSecret'])
+            ->getMock();
+        $api->expects($this->atMost(1))
```

Comment:
@devishke-orange seems not added tests for `create` operation. If you had a issue with dynamic client id creation, mocking is the solution, check this

Reply:
ah thanks. I did try mocking but could not get it to work at that time. will try again